blockstack-vm/src/functions/arithmetic.rs

```diff
@@ -0,0 +1,143 @@
+use super::super::types::Value;
+use super::super::errors::Error;
+use super::super::InterpreterResult;
+
+fn type_force_integer(value: &Value) -> Result<i128, Error> {
+    match *value {
+        Value::Int(int) => Ok(int),
+        _ => Err(Error::TypeError("IntType".to_string(), value.clone()))
+    }
+}
+
+fn binary_comparison<F>(args: &[Value], function: &F) -> InterpreterResult
+where F: Fn(i128, i128) -> bool {
+    if args.len() == 2 {
+        let arg1 = type_force_integer(&args[0])?;
+        let arg2 = type_force_integer(&args[1])?;
+        Ok(Value::Bool((*function)(arg1, arg2)))
+    } else {
+        Err(Error::InvalidArguments("Binary comparison must be called with exactly 2 arguments".to_string()))
+    }
+}
+
+pub fn native_geq(args: &[Value]) -> InterpreterResult {
+    binary_comparison(args, &|x, y| x >= y)
+}
+pub fn native_leq(args: &[Value]) -> InterpreterResult {
+    binary_comparison(args, &|x, y| x <= y)
+}
+pub fn native_ge(args: &[Value]) -> InterpreterResult {
+    binary_comparison(args, &|x, y| x > y)
+}
+pub fn native_le(args: &[Value]) -> InterpreterResult {
+    binary_comparison(args, &|x, y| x < y)
+}
+
+pub fn native_add(args: &[Value]) -> InterpreterResult {
```

Comment:
What happens if `args` has length 0 or 1?  Also, will `args` ever have length > 2?

Reply:
`(+ 1 2 3)` evaluates to 6.
`(+)` returns 0

For native_sub `(-)` errors, `(- 1)` returns the negation of it's single argument, `(- 1 2 3)` return -4.

Multiply and divide behave similarly (`(*)` returns 1, `(/)` is an error)

Reply:
Got it :+1: